src/providers/list/hooks/useProvidersInventoryList.ts

```diff
@@ -0,0 +1,125 @@
+import { useEffect, useRef, useState } from 'react';
+
+import type { ProvidersInventoryList, V1beta1Provider } from '@kubev2v/types';
+import { consoleFetchJSON, useFlag } from '@openshift-console/dynamic-plugin-sdk';
+
+import { getInventoryApiUrl } from '../../../modules/Providers/utils/helpers/getApiUrl';
+import { DEFAULT_FIELDS_TO_AVOID_COMPARING } from '../utils/constants';
+import { k8sGetProvidersByNamespace } from '../utils/k8sGetProvidersByNamespace';
+import { updateInventoryIfChanged } from '../utils/updateInventoryIfChanged';
+
+type UseInventoryParams = {
+  namespace?: string;
+  interval?: number; // Polling interval in milliseconds
+};
+
+type UseInventoryResult = {
+  inventory: ProvidersInventoryList | null;
+  loading: boolean;
+  error: Error | null;
+};
+
+const useProvidersInventoryList = ({
+  interval = 20000,
+  namespace = undefined,
+}: UseInventoryParams): UseInventoryResult => {
+  const [inventory, setInventory] = useState<ProvidersInventoryList | null>(null);
+  const [loading, setLoading] = useState<boolean>(true);
+  const [error, setError] = useState<Error | null>(null);
+  const oldDataRef = useRef<ProvidersInventoryList | null>(null);
+  const oldErrorRef = useRef<string | null>(null);
```

Comment:
why its needed?

Reply:
I though of handling the intervals in case of displaying the same error, but it's not relevant for now - so removed it.
Once refactoring the `InventoryNotReachable` component then we can think of how to handle it since currently the exact error message from backend is not displayed.